### DISCUSSION TITLE
'NoneType' object has no attribute 'mainloop'

### DISCUSSION THREAD

**Discussion by CDJuaum** [TomSchimansky/CustomTkinter#1885] [Q&A] [answered]
I've been working on a project and started using customTkinter on it's earlier versions, up until 4.6.3 it was all good but now, it has a error saying that 'NoneType' object has no attribute 'mainloop', I'm still unsure of what does that mean in this case, because surely my class is not None. 
![image_2023-08-07_172417111](https://github.com/TomSchimansky/CustomTkinter/assets/91756430/80cae323-3526-48fb-b18a-6936a1226c67)
![image_2023-08-07_172424579](https://github.com/TomSchimansky/CustomTkinter/assets/91756430/4c4dee5b-186b-4693-8ee2-097efa43822f)
![image_2023-08-07_172443407](https://github.com/TomSchimansky/CustomTkinter/assets/91756430/4fb784eb-6aed-443c-a657-f560b139b186)

**Comment by avalon60** [comment]
Why not, just put an app.mainloop() instead of the "run"  function call, or a self.mainloop at the end of ___init__?

**Reply by CDJuaum** [reply]
it works the same but I have the decorator on the separate function, it does the same without the "run"

**Reply by CDJuaum** [reply, marked as answer]
nvm found what was causing it, he changed from fg_color to _fg_color, ruined the code lol
